Bottom: 571140dda1ce107830bd5d9349a05cb30a45caca
Top:    ebcff211fbf53f81588a32a8f6bef3bd23feaf41
Author: melsdoerfer <michael@elsdoerfer.com>
Date:   2009-11-04 08:46:41 +0100

Make sure an empty content feed (size/length 0) is flagged as bozo.

Due to a bug in Feed Parser, these cases previously just returned a totally empty result dict.

While not strictly necessary to run FeedPlatform, it is recommended, in order to handle those rare cases were a server sends out an emtpy document.

Reported as issue #149.


---

diff --git a/feedparser/feedparser.py b/feedparser/feedparser.py
index 59e546d..9a341f4 100644
--- a/feedparser/feedparser.py
+++ b/feedparser/feedparser.py
@@ -3416,7 +3416,7 @@ def parse(url_file_stream_or_string, etag=None, modified=None, agent=None, refer
     except Exception, e:
         result['bozo'] = 1
         result['bozo_exception'] = e
-        data = ''
+        data = None
         f = None
 
     # if feed is gzip-compressed, decompress it
@@ -3474,8 +3474,9 @@ def parse(url_file_stream_or_string, etag=None, modified=None, agent=None, refer
             bozo_message = 'no Content-type specified'
         result['bozo'] = 1
         result['bozo_exception'] = NonXMLContentType(bozo_message)
-        
-    result['version'], data, entities = _stripDoctype(data)
+
+    if data is not None:
+        result['version'], data, entities = _stripDoctype(data)
 
     baseuri = http_headers.get('content-location', result.get('href'))
     baselang = http_headers.get('content-language', None)
@@ -3488,7 +3489,7 @@ def parse(url_file_stream_or_string, etag=None, modified=None, agent=None, refer
         return result
 
     # if there was a problem downloading, we're done
-    if not data:
+    if data is None:
         return result
 
     # determine character encoding
